tests/test/paywall.test.js

```diff
@@ -0,0 +1,58 @@
+const url = require('../helpers/url')
+
+jest.setTimeout(30000)
+
+describe('The Unlock Paywall', () => {
+  const testLockAddress = '0x5Cd3FC283c42B4d5083dbA4a6bE5ac58fC0f0267'
+
+  function lockSelector(name) {
+    return `#${name}_${testLockAddress}`
+  }
+
+  beforeAll(async () => {
+    await page.goto(url('/dashboard'))
+    await expect(page).toMatch('Creator Dashboard')
+    await page.waitFor('#CreateLockButton')
+    await page.waitForSelector(`#LockEmbeddCode_${testLockAddress}`)
+    await expect(page).toMatch('30 days')
+    await Promise.all([
+      page.goto(url(`/demo/${testLockAddress}`)),
+      page.waitForNavigation(),
+    ])
+  }, 10000)
+
+  it('should remove the blocker', async () => {
+    await page.waitForFunction(() => !document.querySelector('#_unlock_blocker'))
+  })
+
+  it('should display the lock after the blocker is gone', async () => {
+    await page.waitForFunction(() => window.frames.length)
+    const paywallIframe = page.mainFrame().childFrames()[0]
+    await paywallIframe.waitForSelector(lockSelector('Lock'))
+    await paywallIframe.waitForFunction((ethPriceSelector) => {
+      const eth = document.querySelector(ethPriceSelector)
+      if (!eth) return false
+      return eth.innerText === '0.33 ETH'
+    }, {}, lockSelector('EthPrice'))
+  }, 20000)
```

Comment:
I a bit concerned by these `20000` or `15000` which seem a bit like black magic? Wouldn't i be better to have something consistent or at least defined variables that we can tweak on all tests? How did you come up with these values?

Reply:
I removed them, they were part of debugging initially (same issue as the other thing. Code review process is working well as usual!)